test/validation/api/ipsec/ipsec.h

```diff
@@ -101,9 +110,12 @@ typedef struct {
 		 * differs from that of input test packet (pkt_in).
 		 */
 		uint32_t orig_ip_len;
+		enum ipsec_test_sa_expiry sa_expiry;
```

Comment:
This is part of the "expected output" in ipsec_test_part, but it is not handled exactly like that. Maybe have "IGNORED" value in addition to the "NONE" value and then in the test code always check the expiry status unless the expected status is "IGNORE"?

Reply:
Ack. Further looking at the test, realized that the NONE is not really needed. Replaced the NONE value with IGNORED and updated the test code as suggested.

Reply:
I think the logic and the meaning of the expiry field is still a little bit confusing. I think it would be better if IGNORE would attempt to read status message if they are available and then ignore them, rather than let them pile in the queue from which they get read later when the status changed from IGNORE. Then NONE or similar could still be used for checking that expiry did not happen. The test would then send a bunch of packets with expiry causing a valitation error, then another bunch of packets where expiry is allowed to happen but not required, and finally a third bunch of packets where expiry must happen.

Reply:
I think now I understood what you meant by IGNORE: 
1. For both lookaside and inline cases, we would not ASSERT or return any errors based on the status/result returned.
2. For inline outbound status events, we will consume the status event so that it doesn't pile up.

But I am not quite sure if I understood the meaning for NONE.
I was thinking of having NONE for disabling all SA expiry related logic in the common code. The assumption is that SA expiry is not expected to happen during those tests with NONE as sa_expiry, and it is safe to disable the SA expiry test logic in the common code.

I have updated the tests with these two semantics for IGNORE and NONE. Let me know if this is in alignment with your thoughts.

> The test would then send a bunch of packets with expiry causing a valitation error, then another bunch of packets where expiry is allowed to happen but not required, and finally a third bunch of packets where expiry must happen.

I have updated the test code such that the second loop of test packets would stop once sa_expiry_notified is set. And added another loop where the test packets are sent until it crosses limit + delta. As suggested, this will allow us to check that hard expiry is set for every subsequent packet after sa_expiry_notified is set.